src/Illuminate/Support/Arr.php

```diff
@@ -335,11 +335,11 @@ public static function has($array, $keys)
             }
 
             foreach (explode('.', $key) as $segment) {
-                if (static::accessible($subKeyArray) && static::exists($subKeyArray, $segment)) {
-                    $subKeyArray = $subKeyArray[$segment];
-                } else {
-                    return false;
+                if (!static::accessible($subKeyArray) && !static::exists($subKeyArray, $segment)) {
```

Comment:
I believe you made a mistake here. The code you are looking for is:
```
if (! static::accessible($subKeyArray) || ! static::exists($subKeyArray, $segment)) {
```

Reply:
Uh... I missed to turn the `and` to an `or`!